mpy-cross/main.c

```diff
@@ -242,7 +251,7 @@ MP_NOINLINE int main_(int argc, char **argv) {
                     exit(usage(argv));
                 }
                 a += 1;
-                source_file = argv[a];
+                source_file = backslash_to_forwardslash(argv[a]);
```

Comment:
I was hoping this bit wasn't needed, because it would leave an "escape hatch" where someone could, if they really needed, force the use of backslash in the path name.

But I guess you needed to make this change because in some places in the build it would pass `-s <path-with-backslash>` here, and that needs to be converted.

Reply:
Indeed `source_file` is exactly is the one which needs to be fixed when using make_manifest.py: the argument to mpy-cross looks like `C:\path\to\uasyncio/__init__.py` or similar. Don't think there's a better way?

Reply:
What we're effectively doing now is making a new restriction/definition of .mpy files:
- the source name stored in a .mpy file must use `/` to separate bits of the path

And we need to do this because a lot of the tools (eg `mpy-tool.py` and the way it freezes .mpy files) have implicitly made this assumption.  So now we will enforce it via `mpy-cross`.

I guess that's acceptable then.

The alternative is that the source name in .mpy files can contain `/` and `\` and all tools must handle both characters (what you were trying to fix before).  But that makes the tools complicated.  So I think it's fair enough to make this restriction and use only `/`.

Reply:
> What we're effectively doing now is making a new restriction/definition of .mpy files:

Indeed, but that restriction isn't really 'new' because all tools use it already